ui/js/models/resultsets_store.js

```diff
@@ -275,18 +275,17 @@ treeherder.factory('ThResultSetStore', [
 
         var getAllShownJobs = (pushId) => {
             var shownJobs = [];
-
-            var addIfShown = function (jMap) {
-                if (pushId && jMap.job_obj.result_set_id !== pushId) {
-                    return;
-                }
-                if (jMap.job_obj.visible) {
-                    shownJobs.push(jMap.job_obj);
+            var jobMap = getJobMap();
```

Comment:
This function throws:

    TypeError: job_id.job_obj is undefined

in the console when trying to pin all jobs.  So this is why the selenium tests are failing.

Reply:
Just to be clear, this is when you run the app and click the ``pushpin`` button on the ``PushHeader`` bar.  This button should put all the jobs in that push made available by the current filters into the ``PinBoard`` down below.  Please use this test-case to test your function.  Thanks.

Reply:
OK thanks a lot for your help. It seems to be functioning as intended now.

Reply:
You know, I got curious here, and I think we could simplify this function as:

    const getAllShownJobs = pushId => (
      Object.values(getJobMap())
        .map(jobEl => jobEl.job_obj)
        .filter(job => job.result_set_id === pushId && job.visible)
    );

It could be made a bit more efficient with a  ``reduce`` but this isn't a costly operation, so I think this works pretty well.  How about we go with this?

Reply:
Looks beautiful